src/hubmap_translator.py

```diff
@@ -644,43 +647,58 @@ def generate_doc(self, entity, return_type):
-                                                                       entity['organ'].strip() != '') else None
+                entity['origin_sample'] = copy.copy(entity) if ('specimen_type' in entity) and (entity['specimen_type'].lower() == 'organ') and ('organ' in entity) and (entity['organ'].strip() != '') else None
+                entity['origin_samples'] = copy.copy(entity) if ('specimen_type' in entity) and (entity['specimen_type'].lower() == 'organ') and ('organ' in entity) and (entity['organ'].strip() != '') else None
                 if entity['origin_sample'] is None:
                     try:
                         # The origin_sample is the ancestor which `specimen_type` is "organ" and the `organ` code is set
-                        entity['origin_sample'] = copy.copy(next(a for a in ancestors if ('specimen_type' in a) and (
-                                a['specimen_type'].lower() == 'organ') and ('organ' in a) and (
-                                                                         a['organ'].strip() != '')))
+                        entity['origin_sample'] = copy.copy(next(a for a in ancestors if ('specimen_type' in a) and (a['specimen_type'].lower() == 'organ') and ('organ' in a) and (a['organ'].strip() != '')))
                     except StopIteration:
                         entity['origin_sample'] = {}
+                if entity['origin_samples'] is None:
+                    entity['origin_samples'] = []
+                    for each in ancestors:
+                        if ('specimen_type' in each) and (each['specimen_type'].lower() == 'organ') and ('organ' in each) and (each['organ'].strip() != ''):
+                            entity['origin_samples'].append(each)
 
                 # Remove those added fields specified in `entity_properties_list` from origin_sample and source_sample
                 self.exclude_added_top_level_properties(entity['origin_sample'])
-
+                self.exclude_added_top_level_properties(entity['origin_samples'])
                 # Trying to understand here!!!
                 if entity['entity_type'] == 'Dataset':
                     entity['source_sample'] = None
+                    entity['source_samples'] = None
 
                     e = entity
 
                     while entity['source_sample'] is None:
                         parents = self.call_entity_api(e['uuid'], 'parents')
 
                         try:
-                            # Why?
                             if parents[0]['entity_type'] == 'Sample':
-                                # entity['source_sample'] = parents[0]
                                 entity['source_sample'] = parents
 
                             e = parents[0]
                         except IndexError:
                             entity['source_sample'] = {}
 
+                    e = entity
+                    entity_sdk_instance = EntitySdk(token=self.token, service_url=self.entity_api_url)
```

Comment:
@DerekFurstPitt we need to back out the use of hubmap_sdk, two reasons:
- there's a dependency conflict. hubmap-sdk 1.0.2 depends on urllib3==1.26.7 but the current search-api uses urllib3==1.26.5. The build failed.
- we should still use the `call_entity_api()` method to retrieve the parent entities, for now, to keep the code consistent with the calls used elsewhere. 

Replacing with the hubmap_sdk can be done separately later, and when we do that we should replace everywhere the old calls being used. For the purpose of this PR, we don't want to introduce new changes that are unrepeated.

Reply:
Updated. I understand. I was just trying to save myself some work down the road when inevitably hubmap_sdk gets applied throughout search-api.